src/parse/spec.js

```diff
@@ -28,9 +28,10 @@ define(function(require, exports, module) {
         padding: parsePadding(spec.padding),
         signals: parseSignals(model, spec.signals),
         predicates: parsePredicates(model, spec.predicates),
-        marks: parseMarks(model, spec, width, height),
-        data: parseData(model, spec.data, function() { callback(viewFactory(model)); })
+        marks: parseMarks(model, spec, width, height)
       });
+
+      model.data(parseData(model, spec.data, function() { callback(viewFactory(model)); }))
```

Comment:
Gotcha. Is this (and the similar change made to interactors) necessary to support headless rendering? If not, would you mind reverting these changes and submitting them separately? I would like to keep PRs tightly scoped to each issue to make it easier for me to track all the changes being submitted for consideration. Thanks :)